src/gui/pedalboard/cables/Cable.cpp

```diff
@@ -0,0 +1,105 @@
+#include "Cable.h"
+#include "../BoardComponent.h"
+#include "CableDrawingHelpers.h"
+#include "CableViewConnectionHelper.h"
+#include "CableViewPortLocationHelper.h"
+
+Cable::Cable (const BoardComponent* comp, const CableView& cv, const ConnectionInfo connection) : startProc (connection.startProc), startIdx (connection.startPort), endProc (connection.endProc), endIdx (connection.endPort), cableView (cv), board (comp), cableColour (0xFFD0592C)
+{
+}
+
+Cable::~Cable() = default;
+
+float Cable::getCableThickness (float levelDB)
```

Comment:
On this subject, as you can see I opted to have most of the variables needed for cable drawing included as member variables. Do you have any opinion on if having them all as member variables, or allocating them with each call of the paint function would be better?

Reply:
I don't think there would be a noticeable performance difference... although I would guess having `levelDB` as a local variable might be a little bit faster than the member variable. That said, I think using the member variable is a bit cleaner, so that's probably the right thing to do for now.